src/explore-education-statistics-common/src/validation/serverValidations.ts

```diff
@@ -132,44 +132,76 @@ export function convertServerFieldErrors<FormValues>(
   );
 }
 
-export function isServerValidationError(
+/**
+ * Asserts whether the given error object is an
+ * {@link AxiosError<ServerValidationErrorResponse<T>>}.
+ *
+ * @param error - the error object to check the type of.
+ */
+export function isServerValidationError<T extends string = string>(
   error: unknown,
-  errorMessage?: string,
-): error is AxiosError<ServerValidationErrorResponse> {
+): error is AxiosError<ServerValidationErrorResponse<T>> {
   if (!isAxiosError(error) || !error.response?.data) {
     return false;
   }
 
   const errorDataAsValidationError = error.response
-    .data as ServerValidationErrorResponse;
+    .data as ServerValidationErrorResponse<T>;
 
-  const isServerError =
+  return (
     errorDataAsValidationError.errors !== undefined &&
     errorDataAsValidationError.status !== undefined &&
-    errorDataAsValidationError.title !== undefined;
+    errorDataAsValidationError.title !== undefined
+  );
+}
 
-  if (!errorMessage) {
-    return isServerError;
+/**
+ * Asserts whether the given error object is an
+ * {@link AxiosError<ServerValidationErrorResponse<T>>}.
+ *
+ * See {@link isServerValidationError}
+ *
+ * Additionally, this method checks whether or not the error contains any of
+ * the error messages provided, using {@param fieldName} to determine field
+ * validation, or global validation if {@param fieldName} is omitted.
+ *
+ * If any are included in this error object, this method will return true.
+ *
+ * @param error - the error object to check the type of.
+ * @param errorMessages - array of error messages, any of which can appear
+ * in this error object in order for this method to return true.
+ * @param fieldName - optional fieldName that, if omitted, will be treated
+ * as checking for global errors.
+ */
+export function isAnyServerValidationError<T extends string = string>(
+  error: unknown,
+  errorMessages?: readonly T[],
```

Comment:
Could make this parameter required? In combination with renaming it to `hasErrorMessages`, this method signature would make more sense.

Reply:
Done